File: LeetCode/DataStructure_II/Day4/240_search_a_2d_matrix_2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        pos_x, pos_y = len(matrix)-1, 0
        while True:
            if target < matrix[pos_x][pos_y]:
                pos_x -= 1
            elif target > matrix[pos_x][pos_y]:
                pos_y += 1
            else:
                return True
            if not(0 <= pos_x and pos_x < len(matrix)):
                return False 
            if not(0 <= pos_y and pos_y < len(matrix[0])):
                return False 

File: LeetCode/DayChallenges/DayChallenge_2022_July/240_Search_a_2D_Matrix2.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
 var searchMatrix = function(matrix, target) {
    let [row, col] = [0, matrix[0].length-1];
    while (0 <= row && row < matrix.length && 0 <= col && col < matrix[0].length) {
        if (matrix[row][col] > target) col--;
        else if (matrix[row][col] < target) row ++;
        else {
            return true;
        }
    }
    return false;
};
